### PR TITLE
fix: i18n bindings validation fails for nested *.properties files

### PR DESCRIPTION
**Pull request by mmilko01**
Fix for: https://github.com/SAP/open-ux-tools/issues/2020.

- Checks nested *.properties files when validating key.

### CODE REVIEW

File: packages/preview-middleware-client/src/flp/init.ts [hunk 1]
```diff
@@ -235,7 +255,7 @@ export async function init({
                 }
 
                 if (flexSettings.pluginScript) {
-                    libs.push(pluginScript);
+                    libs.push(pluginScript as string);
```
Comment (nikmace) [inline packages/preview-middleware-client/src/flp/init.ts:258]: Why does this need to be casted to string?

File: packages/preview-middleware-client/src/flp/init.ts [hunk 2]
```diff
@@ -178,19 +179,36 @@ export function registerSAPFonts() {
...
+/**
+ * Create Respource Bundle based on the scenario.
+ *
+ * @param scenario scenario to be used for the resource bundle.
+ */
+export async function loadI18nResourceBundle(scenario: string): Promise<ResourceBundle> {
+    if (scenario === 'ADAPTATION_PROJECT') {
+        const manifest = await getManifestAppdescr();
+        const enhanceWith = (manifest.content as { texts: { i18n: string } }[])
+            .filter((content) => !!content.texts?.i18n)
```
Comment (nikmace) [inline packages/preview-middleware-client/src/flp/init.ts:191]: Do you really need !! here !!content.texts?.i18n ?

File: packages/preview-middleware-client/src/flp/init.ts [hunk 3]
```diff
@@ -178,19 +179,36 @@ export function registerSAPFonts() {
...
+            .map((content) => {
```
Comment (nikmace) [inline packages/preview-middleware-client/src/flp/init.ts:192]: You can return directly

```ts
 .map((content) => ({ bundleUrl: `../${content.texts.i18n}` }));
```

**Review by nikmace — CHANGES_REQUESTED** [review]

File: packages/preview-middleware-client/src/flp/init.ts [hunk 4]
```diff
@@ -178,19 +179,36 @@ export function registerSAPFonts() {
     IconPool.registerFont(suiteTheme);
 }
 
+/**
+ * Create Respource Bundle based on the scenario.
```
[the remark this answers is not in the document]
Comment (mmilko01) [reply packages/preview-middleware-client/src/flp/init.ts:183]: Fixed with: e8c55b97e2800ab6e850e36c5b1a91f62a8be236

File: packages/preview-middleware-client/src/flp/init.ts (hunk 1 above)
Comment (mmilko01) [reply packages/preview-middleware-client/src/flp/init.ts:256]: As discussed due to the usage of `FlexSetting` type, where the pluginScripts property doesn't exist, it resolves to `string | boolean` type. In this case the casting here is needed.

File: packages/preview-middleware-client/src/flp/init.ts (hunk 3 above)
Comment (mmilko01) [reply packages/preview-middleware-client/src/flp/init.ts]: Fixed with: e8c55b97e2800ab6e850e36c5b1a91f62a8be236

**Review by voicis — CHANGES_REQUESTED** [review]

File: packages/preview-middleware-client/src/flp/init.ts [hunk 5]
```diff
@@ -1,11 +1,13 @@
 import Log from 'sap/base/Log';
 import type AppLifeCycle from 'sap/ushell/services/AppLifeCycle';
 import type { InitRtaScript, RTAPlugin, StartAdaptation } from 'sap/ui/rta/api/startAdaptation';
-import type { RTAOptions } from 'sap/ui/rta/RuntimeAuthoring';
+import type { RTAOptions, FlexSettings } from 'sap/ui/rta/RuntimeAuthoring';
+import { default as scenarios, type Scenario } from 'sap/ui/fl/Scenario';
```
[the remark this answers is not in the document]
Comment (mmilko01) [reply packages/preview-middleware-client/src/flp/init.ts]: Fixed with: eb7585363ea751f7376cd5225c08e304160f38ae

**Review by voicis — APPROVED** [review]
- Code adheres to development guidelines
- Review comments addressed
- Good test coverage
- Tested locally

**Review by nikmace — APPROVED** [review]
Code looks good
Coverage is good
Did not test manually

**Review by nikmace — APPROVED** [review]
Code looks good
Test coverage is good
Did not test manually

**Review by voicis — APPROVED** [review]

**Pull request merged**